optuna/cli.py

```diff
@@ -369,6 +373,122 @@ def take_action(self, parsed_args: Namespace) -> None:
             )
 
 
+class _Ask(_BaseCommand):
+    """Create a new trial and suggest parameters."""
+
+    def get_parser(self, prog_name: str) -> ArgumentParser:
+
+        parser = super(_Ask, self).get_parser(prog_name)
```

Comment:
AFAIK, Optuna CLI doesn't provide help messages for subcommands.
(I think it may not be the problem because currently implemented commands are relatively easy to use)

However, arguments of `ask` and `tell` are complicated for me. I want a help message for them.
Please correct me if there is a way to show help in Optuna CLI.

<details>

```sh
> optuna --help
usage: optuna [--version] [-v | -q] [--log-file LOG_FILE] [-h] [--debug] [--storage STORAGE]

optional arguments:
  --version            show program's version number and exit
  -v, --verbose        Increase verbosity of output. Can be repeated.
  -q, --quiet          Suppress output except warnings and errors.
  --log-file LOG_FILE  Specify a file to log output. Disabled by default.
  -h, --help           Show help message and exit.
  --debug              Show tracebacks on errors.
  --storage STORAGE    DB URL. (e.g. sqlite:///example.db)

Commands:
  ask            Create a new trial and suggest parameters.
  complete       print bash completion command (cliff)
  create-study   Create a new study.
  dashboard      Launch web dashboard (beta).
  delete-study   Delete a specified study.
  help           print detailed help for another command (cliff)
  storage upgrade  Upgrade the schema of a storage.
  studies        Show a list of studies.
  study optimize  Start optimization of a study. Deprecated since version 2.0.0.
  study set-user-attr  Set a user attribute to a study.
  tell           Finish a trial created with the ask command.


(venv) 2021-07-27 11:17:09 [~/work/github.com/himkt/optuna] @ cli-ask-tell ●
> optuna ask --help
usage: optuna [--version] [-v | -q] [--log-file LOG_FILE] [-h] [--debug] [--storage STORAGE]

optional arguments:
  --version            show program's version number and exit
  -v, --verbose        Increase verbosity of output. Can be repeated.
  -q, --quiet          Suppress output except warnings and errors.
  --log-file LOG_FILE  Specify a file to log output. Disabled by default.
  -h, --help           Show help message and exit.
  --debug              Show tracebacks on errors.
  --storage STORAGE    DB URL. (e.g. sqlite:///example.db)

Commands:
  ask            Create a new trial and suggest parameters.
  complete       print bash completion command (cliff)
  create-study   Create a new study.
  dashboard      Launch web dashboard (beta).
  delete-study   Delete a specified study.
  help           print detailed help for another command (cliff)
  storage upgrade  Upgrade the schema of a storage.
  studies        Show a list of studies.
  study optimize  Start optimization of a study. Deprecated since version 2.0.0.
  study set-user-attr  Set a user attribute to a study.
  tell           Finish a trial created with the ask command.
```

</details>

Reply:
You're right about the current situation. I think this PR addresses the problem. Do you want to have a look at it? https://github.com/optuna/optuna/pull/2823

Reply:
Thank you so much for the investigation and patching! I approved the PR. 👍